griptape/tokenizers/base_tokenizer.py

```diff
@@ -21,6 +18,16 @@ def count_tokens_left(self, text: str | list) -> int:
         else:
             return 0
 
+    @overload
+    @abstractmethod
+    def count_tokens(self, text: str) -> int:
+        ...
+
+    @overload
+    @abstractmethod
+    def count_tokens(self, text: list[dict]) -> int:
+        ...
+
     @abstractmethod
-    def count_tokens(self, text: str | list) -> int:
+    def count_tokens(self, text: str | list[dict]) -> int:
         ...
```

Comment:
Why do we need both here?

Reply:
Both as in `str` and `list[dict]`? `OpenAiTokenizer` uses a list.

Reply:
Both as in a method with a union type hint _and_ an overload. Is it to make the type checker happy?

Reply:
Ah I see, I can remove the union type hint; is overloading even worth doing here? Originally I had done it to avoid the `isinstance` checks in the subclass methods but I think I had misunderstood python's overloading capabilities.

Reply:
I think it can just be `def count_tokens(self, text: str | list[dict])` without the `@overload` method.

Reply:
Yeah that's what it was originally. Adding `@overload` does give the benefit of multiple method signatures for things like auto complete, but maybe not worth the fork in consistency.

Reply:
Oh, I see! That's a good point about auto complete. Let's have an internal discussion about this since there are multiple places that this will touch.